src/lambdaisland/witchcraft/cursor.clj

```diff
@@ -408,30 +408,36 @@
       (assoc :block-facing dir)
       (assoc :face-direction? false)))
 
+(defn restore-block-facing
+  "Restore the attributes related to block-facing from `old-cursor`."
+  [cursor old-cursor]
+  (merge cursor (select-keys old-cursor [:block-facing :face-direction?])))
```

Comment:
Tiny remark here since this isn't really going to be a hot path, but using select-keys + merge is quite wasteful here, merge especially is known to be kinda slow. I think it's better and more clear to just `(assoc cursor :block-facing (:block-facing old-cursor) :face-direction? (:face-direction? old-cursor))`.